tests/pc/test_retry_count.py

```diff
@@ -0,0 +1,266 @@
+        ConditionalField(ByteField("actor_retry_count_length", 4), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("actor_retry_count", 0), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("actor_retry_count_reserved", "", 1), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count_type", 0x81), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count_length", 4), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count", 0), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("partner_retry_count_reserved", "", 1), lambda pkt:pkt.version == 0xf1),
+        ByteField("terminator_type", 0),
+        ByteField("terminator_length", 0),
+        ConditionalField(XStrFixedLenField("reserved", "", 42), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("reserved", "", 50), lambda pkt:pkt.version != 0xf1),
+    ]
+
+
+@pytest.fixture(scope="class")
+def higher_retry_count_on_peers(request, nbrhosts):
+    if request.config.getoption("neighbor_type") != "sonic":
+        pytest.skip("Only supported with SONiC neighbor")
+
+    featureCheckResult = nbrhosts[nbrhosts.keys()[0]]['host'].shell(
+            "sudo config portchannel retry-count get PortChannel1", module_ignore_errors=True)
+    if featureCheckResult["rc"] != 0:
+        pytest.skip("SONiC neighbor isn't running supported version of SONiC")
+
+    for nbr in list(nbrhosts.keys()):
+        nbrhosts[nbr]['host'].shell("sudo config portchannel retry-count set PortChannel1 5")
+
+    # Wait for retry count info to be updated
+    time.sleep(5)
+
+    yield
+
+    for nbr in list(nbrhosts.keys()):
+        nbrhosts[nbr]['host'].shell("sudo config portchannel retry-count set PortChannel1 3")
+
+    # Wait for retry count info to be updated
+    time.sleep(60)
+
+
+@pytest.fixture(scope="class")
+def higher_retry_count_on_dut(request, duthost):
+    if request.config.getoption("neighbor_type") != "sonic":
+        pytest.skip("Only supported with SONiC neighbor")
+
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+
+    featureCheckResult = duthost.shell("sudo config portchannel retry-count get {}".format(
+        cfg_facts["PORTCHANNEL"].keys()[0]), module_ignore_errors=True)
+    if featureCheckResult["rc"] != 0:
+        pytest.skip("SONiC DUT isn't running supported version of SONiC")
+
+    for port_channel in list(cfg_facts["PORTCHANNEL"].keys()):
+        duthost.shell("sudo config portchannel retry-count set {} 5".format(port_channel))
+
+    # Wait for retry count info to be updated
+    time.sleep(5)
+
+    yield
+
+    for port_channel in list(cfg_facts["PORTCHANNEL"].keys()):
+        duthost.shell("sudo config portchannel retry-count set {} 3".format(port_channel))
+
+    # Wait for retry count info to be updated
+    time.sleep(60)
```

Comment:
No, it takes 60-90 seconds for the retry count to actually get reset to 3, because we have a timer that prevents it from getting reset to 3 after every new version LACPDU is sent.